tests/test_tarfs.py

```diff
@@ -191,12 +191,12 @@ def _build_fixture():
 
             # Create archive
             cmd = ("tar", "-cf", "/dev/stdout", "./file1", "./sub/file2")
-            with subprocess.Popen(cmd, stdout=subprocess.PIPE) as p:
-                stdout, stderr = p.communicate()
-                ec = p.poll()
-                if ec != 0:
-                    raise Exception('tar -cf returned exit code %d', ec)
-                return io.BytesIO(stdout).read()
+            p = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+            stdout, stderr = p.communicate()
```

Comment:
Is there a way you could do this with the `tarfile` module instead? This is going to fail if tested on Windows (no `/dev/stdout`), and I'm also not sure about how platform-specific `tar` binaries may perform, so using `tarfile` should make it consistent across platforms. 

I think if you create a [`TarInfo`](https://docs.python.org/3/library/tarfile.html#tarfile.TarInfo) manually with dotted path and pass it to [`TarFile.addfile`](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.addfile) it should create the same archive.

Reply:
Yes, I'll see what I can do. I didn't think about Windows..